File: CommonRecycler/src/main/java/com/jcodecraeer/xrecyclerview/base/BaseLoadMoreFooter.java
package com.jcodecraeer.xrecyclerview.base;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;

/**
 * Created by guoshuyu on 2017/1/7.
 */

public abstract class BaseLoadMoreFooter extends LinearLayout implements BaseLoadMoreInterface {

    public final static int STATE_LOADING = 0;

    public final static int STATE_COMPLETE = 1;

    public final static int STATE_NOMORE = 2;

    public BaseLoadMoreFooter(Context context) {
        super(context);
    }

    public BaseLoadMoreFooter(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public BaseLoadMoreFooter(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }
}

File: CommonRecycler-Kotlin/src/main/java/com/shuyu/commonrecycler/xrecycler/base/BaseLoadMoreFooter.kt
package com.shuyu.commonrecycler.xrecycler.base

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout

/**
 * Created by guoshuyu on 2017/1/7.
 */

open abstract class BaseLoadMoreFooter : LinearLayout, BaseLoadMoreInterface {

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    companion object {

        const val STATE_LOADING = 0

        const val STATE_COMPLETE = 1

        const val STATE_NOMORE = 2
    }
}
